services/blog.py

```
@@ -356,11 +737,217 @@ def _generate_frontmatter(self, digest: Dict[str, Any]) -> str:
+                        year_month_day = date_obj.strftime("%Y/%m/%d")
+                        return f"{self.blog_base_url}/stories/{year_month_day}/{filename}"
+                    except ValueError:
+                        pass
+        
+        return self.blog_default_image
+    
+
+
+        
+        try:
+            with open(digest_path, 'r') as f:
```

Ruff (0.12.2)

Undefined name `digest_path`

(F821)